### PR TITLE
Do not dereference ExternalForce::_dataSource if it's null.

### PR DESCRIPTION
**Pull request by chrisdembia**
Fixes an issue mentioned in https://github.com/opensim-org/opensim-gui/issues/1035

### Brief summary of changes

- Do not dereference `_dataSource` if it's null. This was occurring if an ExternalLoads' `datafile` was empty.

### Testing I've completed

- Tested that `opensim-cmd` now gives an exception instead of crashing.

### CHANGELOG.md (choose one)

- no need to update because...within 4.0 development

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2350)

### CODE REVIEW

File: OpenSim/Simulation/Model/ExternalForce.cpp [hunk 1]
```diff
@@ -200,10 +200,11 @@ void ExternalForce::extendConnectToModel(Model& model)
     }
 
     if(!_dataSource){
+        throw Exception("ExternalForce: Not Data source has been set.");
```
Comment (tkuchida) [inline OpenSim/Simulation/Model/ExternalForce.cpp]: Suggest "Not Data" -> "No data"

File: OpenSim/Simulation/Model/ExternalForce.cpp [hunk 2]
```diff
@@ -200,10 +200,11 @@ void ExternalForce::extendConnectToModel(Model& model)
...
         // No property set either
-        if((dataSourceProp.size()==0) || (dataSourceProp.getValue(0) == "")){
-            throw(Exception("ExternalForce: Not Data source has been set."));
-        }
+        // if((dataSourceProp.size()==0) || (dataSourceProp.getValue(0) == "") ||
```
Comment (tkuchida) [inline OpenSim/Simulation/Model/ExternalForce.cpp]: Perhaps just delete these lines?

**Comment by chrisdembia** [comment]
@tkuchida I addresed your comments. Thanks for the review.

**Comment by aymanhab** [comment]
@chrisdembia I assume the issue here is that the old property was not returning empty but a value of "". If that' the case according to your testing then let's merge this. Thank you.

**Comment by chrisdembia** [comment]
The old property was `"Unassigned"`. I'll let you merge this.

**Pull request merged**